packages/react/src/components/TreeView/TreeNode.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { CaretDown16 } from '@carbon/icons-react';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, match, matches } from '../../internal/keyboard';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function TreeNode({
```

Comment:
Quick style-nit:

- export default goes after `propTypes`
- For unstable code we should do [`unstable_ComponentName` as the name](https://github.com/carbon-design-system/carbon/blob/master/docs/style.md#naming-experimental-code)

Reply:
this naming scheme will conflict with the rules we have for React hooks in our ESLint config though

Reply:
yeah it looks like unlike the document you linked, we are currently exporting with `Unstable_ComponentName` convention (probably because of the ESLint errors I mentioned previously) so we may need to update the style guide

Reply:
@emyarod ah got it, what's the rule that you're running into with the lint error?

Reply:
It is the rules of hooks one? 👀

Reply:
Opened up a PR to update: https://github.com/carbon-design-system/carbon/pull/6642

Reply:
I don't recall which rule it was exactly but it was about having uncapitalized component names